src/plugins/discover/public/hooks/use_navigation_props.tsx

```diff
@@ -67,6 +60,39 @@ const getStateParams = ({
   };
 };
 
+/**
+ * Use callback handler on click. Otherwise href is used.
+ */
+const useButtonNavigationRef = ({ onClick }: { onClick: () => void }) => {
```

Comment:
I think we might be able to avoid manually managing these event listeners. It looks like EUI buttons accept both `href` and `onClick` props, and we can prevent the `onClick` if a modifier is held. The linter doesn't like it, but we can disable the rule and everything seems to work as intended: https://codesandbox.io/s/nervous-payne-rd99o7?file=/demo.js. I think it would clean up this hook a bit, what do you think?

Also a couple links to the EUI docs I found while looking into this:
- https://elastic.github.io/eui/#/navigation/button#buttons-as-links
- https://github.com/elastic/eui/blob/main/wiki/react-router.md#react-router-51

Reply:
Thx for finding, applied your approach with both `onClick` and `href`.